include/system.h

```diff
@@ -263,28 +264,39 @@ typedef struct
 } stdio_t;
 
 /**
- * @brief Register a filesystem with newlib
- *
- * This function will take a prefix in the form of 'prefix:/' and a pointer
- * to a filesystem structure of relevant callbacks and register it with newlib.
- * Any standard open/fopen calls with the registered prefix will be passed
- * to this filesystem.  Userspace code does not need to know the underlying
- * filesystem, only the prefix that it has been registered under.
+ * @brief Time hook structure
  *
- * The filesystem pointer passed in to this function should not go out of scope
- * for the lifetime of the filesystem.
+ * This structure provides optional callback hooks for code wishing to
+ * implement C time functions.  Any function that code does not wish to handle
+ * should be left as a NULL pointer.
+ */
+typedef struct
+{
+    /** 
+     * @brief Function to call to retrieve the current date/time
+     * 
+     * @return number of seconds since 1970-01-01 00:00:00 UTC
+     */
+    time_t (*gettime)( void );
+    /** 
+     * @brief Function to call to set the current date/time
+     * 
+     * @param time number of seconds since 1970-01-01 00:00:00 UTC
+     * 
+     * @return whether the time was set successfully
+     */
+    bool (*settime)( time_t );
+} time_hooks_t;
+
+/**
+ * @brief Register a filesystem with newlib
  *
  * @param[in] prefix
- *            Prefix of the filesystem
+ *            The prefix that was used to register the filesystem
  * @param[in] filesystem
- *            Structure of callbacks for various functions in the filesystem.
- *            If the registered filesystem doesn't support an operation, it
- *            should leave the callback null.
- * 
- * @retval -1 if the parameters are invalid
- * @retval -2 if the prefix is already in use
- * @retval -3 if there are no more slots for filesystems
- * @retval 0 if the filesystem was registered successfully
+ *            Pointer to structure containing filesystem callbacks
+ *
+ * @return 0 on successful registration or a negative value on failure.
```

Comment:
Why dropping the description of the error codes here?

Reply:
Not sure how that happened; reverted https://github.com/DragonMinded/libdragon/pull/563/commits/97ce533d4f0a9df128de2856be7514d3ce95409c